share/lib/python/neuron/coreneuron.py

```diff
@@ -0,0 +1,76 @@
+from neuron import h
+
+# Properties settable by user
+enable = False     # Use CoreNEURON when calling ParallelContext.psolve(tstop).
+gpu = False        # Activate GPU computation.
+cell_permute = 1   # 0 no permutation; 1 optimize node adjacency
+                   # 2 optimize parent node adjacency (only for gpu = True)
+warp_balance = 0   # Number of warps to balance. (0 no balance)
```

Comment:
A warp is the number of threads executed simultaneously by a streaming multiprocessor (often 32).  The argument sent to  CoreNEURON is --nwarp integer where 0 means not to balance. The balance talked about in coreneuron is not within warp but between warp. i.e an individual warp is always balanced as much as possible between its threads. 
"Between warp" balance means
```
 // use LPT algorithm to balance cells so all warps have similar number
// of compartments.
// NB: Ideally we'd balance so that warps have similar ncycle. But we do not
// know how to predict warp quality without an apriori set of cells to
// fill the warp. For large numbers of cells in a warp,
// it is a justifiable speculation to presume that there will be very
// few holes in warp filling. I.e., ncycle = ncompart/warpsize
```
Actually, I need to study the relevant coreneuron code and add comments. I can't vouch for what it is doing.
There is nothing in the code that depends on typical warp size 32 which seems odd to me.

Reply:
How about calling it "thread_balance"?